devito/propagator.py

```diff
@@ -267,6 +267,22 @@ def sympy_to_cgen(self, stencils):
 
         return cgen.Block(factors+kernel)
 
+    def expr_dtype(self, expr):
+        """Gets the resulting dtype of an expression.
+
+        :returns: The dtype. Defaults to `np.float64` if none found.
+        """
+        dtypes = []
+
+        for e in preorder_traversal(expr):
+            if hasattr(e, 'dtype'):
+                dtypes.append(e.dtype)
+
+        if dtypes:
+            return np.find_common_type(dtypes, [])
+        else:
+            return np.float64
```

Comment:
Does the Propagator not have a `dtype` argument? Should we not be defaulting to this one?